acceptance: disable hyper-v time sync for nightlies

cuongdo opened the pull request:
hyper-v time synchronization appears to be the cause of various issues,
including some recent failures of TestRebalance_3To5Small and the
requirement that the max clock offset be set to the rather high 1 second
for our nightlies. This change disables the faulty time sync by using
the same approach the prod scripts use.

Commit:
acceptance: disable hyper-v time sync for nightlies

hyper-v time synchronization appears to be the cause of various issues,
including some recent failures of TestRebalance_3To5Small and the
requirement that the max clock offset be set to the rather high 1 second
for our nightlies. This change disables the faulty time sync by using
the same approach the prod scripts use.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/16155)
<!-- Reviewable:end -->

tschottdorf commented on pkg/acceptance/terraform/azure/disable-hyperv:
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
```
No `set -euo pipefail`?

tschottdorf commented on pkg/acceptance/terraform/azure/disable-hyperv:
```diff
@@ -0,0 +1,43 @@
+# Stop ntp, force ntp sync, restart ntp.
+sudo service ntp stop
+sudo ntpdate -b ntp.ubuntu.com
+sudo service ntp start
```
After the restart, what ntp server is it going to use?

tschottdorf commented on pkg/acceptance/terraform/azure/main.tf:
```diff
@@ -225,35 +230,34 @@ FILE
+      "sudo service supervisor stop",
       "if [ ! -e supervisor.pid ]; then supervisord -c supervisor.conf; fi",
+      # Disable hypervisor clock sync, because it can cause an unrecoverable
+      # amount of clock skew. This also forces an NTP sync
```
nit: missing punctuation.

tamird commented:
Does this mean we can remove the setting of max offset = 1 second?

---

Reviewed 1 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[pkg/acceptance/terraform/azure/disable-hyperv, line 14 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-Kl0ankO_pDEDy1sQUxs:-Kl0ankPRwhYnF7sow__:b-d4bmds) ([raw file](https://github.com/cockroachdb/cockroach/blob/6129524eea3259e2020f1c27f7ced047b18c4b6f/pkg/acceptance/terraform/azure/disable-hyperv#L14)):*
> ```
> 
> # According to lsvmbus, the time sync class ID is:
> TIME_SYNC_CLASS_ID="{9527e630-d0ae-497b-adce-e80ab0175caf}"
> ```

I realize this file is probably copied from the production repo, but there are a bunch of shell oddities here. For instance, I believe the quotes here are unnecessary. If you don't mind, I'll push a commit addressing these nits so you don't have to worry about them!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-:-Kl0bhLLbovgGdupIRRm:bsj2xml)*
<!-- Sent from Reviewable.io -->

cuongdo commented:
Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[pkg/acceptance/terraform/azure/disable-hyperv, line 2 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-Kl0XbkrZefB05Q-M5CH-r1-2:-KlOowmmvgi59Mg2CGc9:bkrqke3) ([raw file](https://github.com/cockroachdb/cockroach/blob/6129524eea3259e2020f1c27f7ced047b18c4b6f/pkg/acceptance/terraform/azure/disable-hyperv#L2)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

No `set -euo pipefail`?
</blockquote></details>

This was lifted from `crl-prod` with minimal changes. Looks like Tamir fixed this in his revision.

---

*[pkg/acceptance/terraform/azure/disable-hyperv, line 14 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-Kl0ankO_pDEDy1sQUxs:-KlPGnxQjaxRIjaTwz7E:bs8s4px) ([raw file](https://github.com/cockroachdb/cockroach/blob/6129524eea3259e2020f1c27f7ced047b18c4b6f/pkg/acceptance/terraform/azure/disable-hyperv#L14)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

I realize this file is probably copied from the production repo, but there are a bunch of shell oddities here. For instance, I believe the quotes here are unnecessary. If you don't mind, I'll push a commit addressing these nits so you don't have to worry about them!
</blockquote></details>

Thanks! Less shell scripting is always fine by me.

---

*[pkg/acceptance/terraform/azure/disable-hyperv, line 43 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-Kl0XbkrZefB05Q-M5CH-r1-43:-KlOr4pUHHLdGaSxQUMG:b-xuo6kx) ([raw file](https://github.com/cockroachdb/cockroach/blob/6129524eea3259e2020f1c27f7ced047b18c4b6f/pkg/acceptance/terraform/azure/disable-hyperv#L43)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

After the restart, what ntp server is it going to use?
</blockquote></details>

These are the defaults in `ntp.conf`:

```
# Use servers from the NTP Pool Project. Approved by Ubuntu Technical Board
# on 2011-02-08 (LP: #104525). See http://www.pool.ntp.org/join.html for
# more information.
pool 0.ubuntu.pool.ntp.org iburst
pool 1.ubuntu.pool.ntp.org iburst
pool 2.ubuntu.pool.ntp.org iburst
pool 3.ubuntu.pool.ntp.org iburst

# Use Ubuntu's ntp server as a fallback.
pool ntp.ubuntu.com
```

---

*[pkg/acceptance/terraform/azure/main.tf, line 258 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-Kl0XbkrZefB05Q-M5CG-r1-258:-KlOqI_rDx9COJtq3o4F:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6129524eea3259e2020f1c27f7ced047b18c4b6f/pkg/acceptance/terraform/azure/main.tf#L258)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

nit: missing punctuation.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155)*
<!-- Sent from Reviewable.io -->

tamird commented:
@cuongdo can you address my question above?

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> otherwise, though.

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-:-KlPVyaQjlI_uswhSpgH:boevrds)*
<!-- Sent from Reviewable.io -->

cuongdo commented:
Yes, I'm changing the nightlies to use the default max offset (500 ms).

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-:-KlPYqI_sSxJd3n4DyEp:b-j1wj4v)*
<!-- Sent from Reviewable.io -->

tamird commented:
Great, thank you.

On Tue, May 30, 2017 at 2:13 PM, Cuong Do <notifications@github.com> wrote:

> Yes, I'm changing the nightlies to use the default max offset (500 ms).
> ------------------------------
>
> *Comments from Reviewable
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/16155#-:-KlPYqI_sSxJd3n4DyEp:b-j1wj4v>*
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/16155#issuecomment-304962500>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPMfXsRpYtyE9OSEo6RMw7YZ9n5bkks5r_Fw8gaJpZM4Nm7Bt>
> .
>
